### PR TITLE
repo: use is_repo_import to check for imports in `outs_graph`

### LINKED ISSUE #9912: unnecessary remote repository fetches
DVC has started cloning external repositories even on cases where it's not strictly required. 
Anything that uses `outs_graph` has started cloning external repositories.

Commands like `add/plots modify/pull/push/fetch/move` are affected, and maybe potentially others.

### Reproduce

```bash
git clone https://github.com/iterative/example-get-started.git
cd example-get-started
dvc add foo
```

### Expected

No cloning occurs.

### PR DESCRIPTION
**Pull request by pmrowla**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Fixes #9912

### CODE REVIEW

File: dvc/repo/graph.py [hunk 1]
```diff
@@ -158,10 +158,9 @@ def build_outs_graph(graph, outs_trie):
 
     outs_graph.add_nodes_from(outs_trie.values())
     for stage in graph.nodes():
+        if stage.is_repo_import:
+            continue
         for dep in stage.deps:
-            if dep.fs_path is None:
-                # RepoDependency don't have a path
```
Comment (pmrowla) [inline dvc/repo/graph.py:163]: This is no longer valid since repo deps now have a dvcfs fs_path

File: dvc/repo/graph.py [hunk 2]
```diff
@@ -158,10 +158,9 @@ def build_outs_graph(graph, outs_trie):
 
     outs_graph.add_nodes_from(outs_trie.values())
     for stage in graph.nodes():
+        if stage.is_repo_import:
+            continue
```
Comment (pmrowla) [inline dvc/repo/graph.py:162]: I think this check could actually be `if stage.is_import` here since outs can't overlap with external dep paths now?

(But I left the check as repo imports only for now to match the original behavior, and since import-url deps won't cause the erepo git cloning issue)

**Review by skshetry — APPROVED** [review]

**Comment by skshetry** [comment]
Thanks @pmrowla.

**Review by shcheklein — COMMENTED** [review]
Thanks @pmrowla for the quick fix! Curious what is your take on adding tests on issues like this (and probably for bug fixes in general?). (I had a conversation with @efiop and our opinions differ)- I wanted to calibrate with you and probably with the whole team as well :) (clearly it goes case by case, and there are very valid exception from any policy).

**Comment by pmrowla** [comment]
I'm not really sure that a test for this would be particularly useful. Everything here was behaving as intended. The unwanted behavior was introduced because the underlying support for repo imports has been improved (so they can now be treated like a regular dependency that has a path). At the end of the day, both `dvc add` and the `outs_graph` were working correctly both before and after this patch.

Basically this is something that I would expect to be caught via higher level QA and/or benchmarks, not not necessarily something that needs to be tested at the DVC func or unit test level.

**Comment by skshetry** [comment]
I see this as a design issue with the current `fs.path` thing. It should not be this easy to have unwanted side effects (especially on a DVCS and a build system where you need controlled side-effects).
